src/server/pkg/sync/sync.go

```diff
@@ -105,22 +125,34 @@ func (p *Puller) makeFile(path string, f func(io.Writer) error) (retErr error) {
 			retErr = err
 		}
 	}()
-	return f(file)
+	w := &sizeWriter{w: file}
+	if err := f(w); err != nil {
+		return err
+	}
+	atomic.AddInt64(&p.size, w.size)
+	return nil
 }
 
 // Pull clones an entire repo at a certain commit.
 // root is the local path you want to clone to.
 // fileInfo is the file/dir we are puuling.
 // pipes causes the function to create named pipes in place of files, thus
 // lazily downloading the data as it's needed.
-func (p *Puller) Pull(client *pachclient.APIClient, root string, repo, commit, file string, pipes bool, concurrency int) error {
+func (p *Puller) Pull(client *pachclient.APIClient, root string, repo, commit, file string,
+	pipes bool, concurrency int, tree hashtree.OpenHashTree, treeRoot string) error {
```

Comment:
Will comment, `tree` is an object which the Pulled data gets mirrored into. `treeRoot` is the path to use as the root for that mirror.